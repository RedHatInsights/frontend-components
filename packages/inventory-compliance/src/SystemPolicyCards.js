import React from 'react';
import { Grid, GridItem, Card, CardBody } from '@patternfly/react-core';
import SystemPolicyCard from './SystemPolicyCard';
import propTypes from 'prop-types';
import { Instagram } from 'react-content-loader';

class SystemPolicyCards extends React.Component {
    systemPolicyCards() {
        const { policies } = this.props;
        return policies.filter((policy) => (policy.rulesFailed + policy.rulesPassed) > 0).map(
            (policy, i) =>
                <GridItem sm={ 12 } md={ 12 } lg={ 6 } xl={ 4 } key={ i }>
                    <SystemPolicyCard policy={ policy } />
                </GridItem>
        );
    }

    render() {
        const { loading } = this.props;

        return (
            <React.Fragment>
                <Grid gutter='md'>
                    { this.systemPolicyCards() }
                    { loading && [ ...Array(3) ].map((_item, i) => (
                        <GridItem span={ 4 } key={ i }>
                            <Card>
                                <CardBody>
                                    <Instagram />
                                </CardBody>
                            </Card>
                        </GridItem>
                    )) }
                </Grid>
            </React.Fragment>
        );
    }
}

SystemPolicyCards.propTypes = {
    policies: propTypes.array,
    loading: propTypes.bool
};

SystemPolicyCards.defaultProps = {
    policies: []
};

export default SystemPolicyCards;
