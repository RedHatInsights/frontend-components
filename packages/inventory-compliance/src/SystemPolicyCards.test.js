import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SystemPolicyCards from './SystemPolicyCards';
import { IntlProvider } from 'react-intl';

describe('SystemPolicyCards component', () => {
    const policies = [{
        rulesPassed: 30,
        rulesFailed: 10,
        lastScanned: '2019-03-06T06:20:13Z',
        refId: 'xccdf_org.ssgproject.content_profile_pci-dss',
        name: 'PCI-DSS v3 Control Baseline for Red Hat Enterprise Linux 7',
        compliant: false
    }];

    it('should render loading state', () => {
        const wrapper = shallow(
            <IntlProvider locale={ navigator.language }>
                <SystemPolicyCards policies={ policies } loading={ true } />
            </IntlProvider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render real table', () => {
        const wrapper = shallow(
            <IntlProvider locale={ navigator.language }>
                <SystemPolicyCards policies={ policies } loading={ false } />
            </IntlProvider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
