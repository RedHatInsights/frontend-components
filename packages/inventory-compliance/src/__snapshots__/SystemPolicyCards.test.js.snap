// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemPolicyCards component should render loading state 1`] = `
<SystemPolicyCards
  loading={true}
  policies={
    Array [
      Object {
        "compliant": false,
        "lastScanned": "2019-03-06T06:20:13Z",
        "name": "PCI-DSS v3 Control Baseline for Red Hat Enterprise Linux 7",
        "refId": "xccdf_org.ssgproject.content_profile_pci-dss",
        "rulesFailed": 10,
        "rulesPassed": 30,
      },
    ]
  }
/>
`;

exports[`SystemPolicyCards component should render real table 1`] = `
<SystemPolicyCards
  loading={false}
  policies={
    Array [
      Object {
        "compliant": false,
        "lastScanned": "2019-03-06T06:20:13Z",
        "name": "PCI-DSS v3 Control Baseline for Red Hat Enterprise Linux 7",
        "refId": "xccdf_org.ssgproject.content_profile_pci-dss",
        "rulesFailed": 10,
        "rulesPassed": 30,
      },
    ]
  }
/>
`;
